server/src/test/java/io/crate/integrationtests/GCDanglingArtifactsIntegrationTest.java

```diff
@@ -0,0 +1,83 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.integrationtests;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.elasticsearch.cluster.metadata.IndexMetadata.SETTING_INDEX_UUID;
+import static org.elasticsearch.cluster.metadata.IndexMetadata.SETTING_INDEX_VERSION_CREATED;
+import static org.elasticsearch.cluster.metadata.IndexMetadata.SETTING_NUMBER_OF_REPLICAS;
+import static org.elasticsearch.cluster.metadata.IndexMetadata.SETTING_NUMBER_OF_SHARDS;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateUpdateTask;
+import org.elasticsearch.cluster.ack.ClusterStateUpdateResponse;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Priority;
+import org.elasticsearch.common.UUIDs;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.IntegTestCase;
+import org.junit.Test;
+
+import io.crate.action.FutureActionListener;
+
+public class GCDanglingArtifactsIntegrationTest extends IntegTestCase {
+
+    @Test
+    public void test_remove_dangling_indices() throws Exception {
+        String danglingIndexName = ".tmp.dangling.index";
+        FutureActionListener<ClusterStateUpdateResponse> future = new FutureActionListener<>();
+        cluster().getCurrentMasterNodeInstance(ClusterService.class)
+            .submitStateUpdateTask("test-create-dangling-index", new ClusterStateUpdateTask(Priority.HIGH) {
+                @Override
+                public ClusterState execute(ClusterState currentState) throws Exception {
+                    IndexMetadata indexMetadata = IndexMetadata.builder(danglingIndexName)
+                        .settings(Settings.builder()
+                            .put(SETTING_INDEX_UUID, UUIDs.randomBase64UUID())
+                            .put(SETTING_NUMBER_OF_SHARDS, 2)
+                            .put(SETTING_NUMBER_OF_REPLICAS, 0)
+                            .put(SETTING_INDEX_VERSION_CREATED.getKey(), Version.CURRENT)
+                            .build())
+                        .build();
+                    Metadata.Builder mdBuilder = Metadata.builder(currentState.metadata());
+                    mdBuilder.put(indexMetadata, false);
```

Comment:
Could we also add a `Table` to verify the `allIndices.contains()` logic?
Maybe also a `BlogTable` to verify the new `relations()` logic.